csharp/src/Ice/StreamWrapper.cs

```diff
@@ -18,24 +18,22 @@ namespace IceInternal
     // For output streams, we use a different strategy:
     // Slice sequences are encoded on the wire as a count of elements, followed
     // by the sequence contents. For arbitrary .NET classes, we do not know how
-    // big the sequence that is eventually written will be. To avoid excessive
-    // data copying, this class maintains a private _bytes array of 254 bytes and,
-    // initially, writes data into that array. If more than 254 bytes end up being
-    // written, we write a dummy sequence size of 255 (which occupies five bytes
-    // on the wire) into the stream and, once this class is disposed, patch
-    // that size to match the actual size. Otherwise, if the _bytes buffer contains
+    // big the sequence will be. To avoid excessive data copying, this class maintains
+    // a private _bytes array of 254 bytes and, initially, writes data into that array.
+    // If more than 254 bytes end up being written, we write a dummy sequence size of 255
+    // (which occupies five bytes on the wire) into the stream and, once this class is disposed,
+    // patch that size to match the actual size. Otherwise, if the _bytes buffer contains
     // fewer than 255 bytes when this class is disposed, we write the sequence size
     // as a single byte, followed by the contents of the _bytes buffer.
-    //
-
-    public class OutputStreamWrapper : Stream
+    internal class OutputStreamWrapper : Stream
```

Comment:
Is there a compelling reason to not simply merge these wrappers into InputStream and OutputStream?

Both InputSteam and OutputStream could derive from Stream. This way, knowledge about size encoding and rewrites could remain private to OutputStream/InputStream.

Reply:
I think main reason is that System.IO.Stream will become part of the public API, but we only support a subset of it, specially the way write is implemented to write an Ice byte sequence.

To me makes more sense to keep this internal wrappers